### ISSUE TITLE
loop-var-prefix rule breaks COP rule 3.1.4

### ISSUE THREAD

**Issue by alice-rc** [ansible/ansible-lint#2838]
##### Summary

The loop-var-prefix expects the variable to start with the role name. This is not correct as loop variables certainly fall under the COP 3.1.4 rule: **Moreover, internal variables (those that are not expected to be set by users) are to be prefixed by two underscores: __foo_variable**
So this loop-var test should be checking for __variable_name NOT $ROLE_variable_name. If you wanted to go way beyond the standard (which should be optional) then the test should be for __$ROLE_variable_name.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint 6.10.0 using ansible 2.13.4
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

```console (paste below)
ansible-lint output:
roles/common/tasks/main.yml:5: loop-var-prefix[wrong]: Loop variable should start with common_ (warning)

```

##### Desired Behavior

Should not report an error or warning when the loop variable begins with two underscores

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

Reports a warning when the loop variable begins with two underscores

Please give some details of what is happening.
Include a [minimum complete verifiable example] with:

- minimized playbook to reproduce the error
- the output of running ansible-lint including the command line used
- if you're getting a stack trace, also the output of
  `ansible-playbook --syntax-check playbook`

```paste below
Role task file excerpt:

loop_control:
    loop_var: __common_loop
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by cidrblock** [comment]
@alice-rc This is a _awesome_catch. TY.  The regex can be made to match either $ROLE or __$ROLE until such time we can switch fully to __$ROLE (this would be a significant/major/breaking change.
